File: src/main/java/com/nao4j/subtrol/service/slice/SubscriptionSliceService.java
package com.nao4j.subtrol.service.slice;

import com.nao4j.subtrol.document.internal.ExactPeriod;
import com.nao4j.subtrol.document.internal.Quantity.QuantityType;
import com.nao4j.subtrol.document.internal.Subscription;

import java.util.Collection;
import java.util.Set;

public interface SubscriptionSliceService {

    Collection<ExactPeriod> slice(Subscription subscription, ExactPeriod period);

    Set<QuantityType> dimensions();

}

File: src/main/kotlin/com/nao4j/subtrol/service/slice/SubscriptionSliceService.kt
package com.nao4j.subtrol.service.slice

import com.nao4j.subtrol.model.ExactPeriod
import com.nao4j.subtrol.model.Subscription

interface SubscriptionSliceService {

    fun slice(subscription: Subscription, period: ExactPeriod): Collection<ExactPeriod>

}
